### PR TITLE
Bump default VPA version to 0.14.0

### PR DESCRIPTION
**Pull request by kgolab**
#### What type of PR is this?

/kind cleanup
/kind documentation

Actually it's part of the release process.

#### What this PR does / why we need it:

The first step of the [finalize release section](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/RELEASE.md#finalize-release).

The release process itself is tracked under #5851.

### CODE REVIEW

File: vertical-pod-autoscaler/README.md [hunk 1]
```diff
@@ -48,12 +48,13 @@ procedure described below.
...
 # Installation
 
-The current default version is Vertical Pod Autoscaler 0.13.0
+The current default version is Vertical Pod Autoscaler 0.14.0
 
 ### Compatibility
 
 | VPA version     | Kubernetes version |
 |-----------------|--------------------|
+| 0.14            | 1.25+              |
```
Comment (jbartosik) [inline vertical-pod-autoscaler/README.md:57]: Did we test with 1.25?

**Review by jbartosik — COMMENTED** [review]
Mostly looks good to me, I have 1 question

File: vertical-pod-autoscaler/README.md (hunk 1 above)
Comment (kgolab) [reply vertical-pod-autoscaler/README.md:57]: Yes, following the procedure directly meant that I used GKE cluster with version 1.25.8-gke.1000 (both KCP & nodes).

**Review by jbartosik — APPROVED** [review]
/lgtm
/approve

**Pull request merged**